services/fileServices/MdPageServices/__tests__/ContactUsPageService.spec.js

```diff
@@ -0,0 +1,171 @@
+    update: jest.fn(),
+  }
+
+  jest.mock("@utils/markdown-utils", () => ({
+    retrieveDataFromMarkdown: jest.fn().mockReturnValue({
+      frontMatter: mockFrontMatter,
+      pageContent: mockContent,
+    }),
+    convertDataToMarkdown: jest.fn().mockReturnValue(mockRawContactUsContent),
+  }))
+
+  const {
+    ContactUsPageService,
+  } = require("@services/fileServices/MdPageServices/ContactUsPageService")
+  const service = new ContactUsPageService({
+    gitHubService: mockGithubService,
+    footerYmlService: mockFooterYmlService,
+  })
+
+  const {
+    retrieveDataFromMarkdown,
+    convertDataToMarkdown,
+  } = require("@utils/markdown-utils")
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+  mockFooterYmlService.read.mockResolvedValue({
+    content: mockFooterContent,
+    sha: mockFooterSha,
+  })
+
+  describe("Read", () => {
+    mockGithubService.read.mockResolvedValue({
+      content: mockRawContactUsContent,
+      sha: mockContactUsSha,
+    })
+
+    it("Reading the contact us page works correctly", async () => {
+      await expect(service.read(reqDetails)).resolves.toMatchObject({
+        content: { frontMatter: mockFrontMatter, pageBody: mockContent },
+        sha: mockContactUsSha,
+      })
+
+      expect(retrieveDataFromMarkdown).toHaveBeenCalledWith(
+        mockRawContactUsContent
+      )
+      expect(mockGithubService.read).toHaveBeenCalledWith(reqDetails, {
+        fileName: CONTACT_US_FILE_NAME,
+        directoryName: CONTACT_US_DIRECTORY_NAME,
+      })
+      expect(mockFooterYmlService.read).toHaveBeenCalledWith(reqDetails)
+    })
+
+    it("Propagates the correct error on failed retrieval", async () => {
+      mockFooterYmlService.read.mockRejectedValueOnce(new NotFoundError(""))
+
+      await expect(service.read(reqDetails)).rejects.toThrowError(NotFoundError)
+
+      expect(retrieveDataFromMarkdown).toHaveBeenCalledWith(
+        mockRawContactUsContent
+      )
+      expect(mockGithubService.read).toHaveBeenCalledWith(reqDetails, {
+        fileName: CONTACT_US_FILE_NAME,
+        directoryName: CONTACT_US_DIRECTORY_NAME,
+      })
+      expect(mockFooterYmlService.read).toHaveBeenCalledWith(reqDetails)
+    })
+  })
+
+  describe("Update", () => {
+    const oldSha = "54321"
+    const updatedFeedback = "updated"
+    mockGithubService.update.mockResolvedValue({ newSha: mockContactUsSha })
+    it("Updating page content works correctly", async () => {
+      const mockUpdatedFrontMatter = {
+        ...mockContactUsContent.frontMatter,
+        feedback: updatedFeedback,
+      }
+
+      await expect(
+        service.update(reqDetails, {
+          fileName: CONTACT_US_FILE_NAME,
+          content: mockContent,
+          frontMatter: mockUpdatedFrontMatter,
+          sha: oldSha,
+        })
+      ).resolves.toMatchObject({
+        content: { frontMatter: mockUpdatedFrontMatter, pageBody: mockContent },
+        oldSha,
+        newSha: mockContactUsSha,
+      })
```

Comment:
i think this got abit hard to read; could we perhaps separate the action (`service.update`) from the expectation (`expect(...).resolves.toMatchObject(...)`)

Reply:
I've split up the req and resp object definitions so we can still use `.resolves/.rejects` https://github.com/isomerpages/isomercms-backend/pull/349/commits/5e6d1148ec52c0c2fec2e9e3cd423be26e34b378